File: backend/create_admin.py
"""Script para criar um usuário admin no banco de dados MongoDB"""
import asyncio
from motor.motor_asyncio import AsyncIOMotorClient
from passlib.context import CryptContext
from datetime import datetime, timezone
import uuid
import os
from dotenv import load_dotenv

load_dotenv()

# Configuração
pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")

async def create_admin_user():
    # Conectar ao MongoDB
    mongo_url = os.getenv("MONGODB_URL", "mongodb://localhost:27017")
    client = AsyncIOMotorClient(mongo_url)
    db = client.galpao

    # Dados do admin
    admin_data = {
        "id": str(uuid.uuid4()),
        "email": "admin@nextfit.com",
        "nome": "Administrador",
        "role": "admin",
        "ativo": True,
        "senha_hash": pwd_context.hash("admin123"),
        "criado_em": datetime.now(timezone.utc).isoformat()
    }

    # Verificar se já existe
    existing = await db.users.find_one({"email": admin_data["email"]})
    if existing:
        print(f"Usuário {admin_data['email']} já existe!")
        return

    # Criar usuário
    await db.users.insert_one(admin_data)
    print("=" * 50)
    print("Usuário Admin Criado com Sucesso!")
    print("=" * 50)
    print(f"Email: {admin_data['email']}")
    print(f"Senha: admin123")
    print(f"Role: {admin_data['role']}")
    print("=" * 50)

if __name__ == "__main__":
    asyncio.run(create_admin_user())

File: backend/create_admin.js
// Script para criar usuário admin via API
const http = require('http');

const data = JSON.stringify({
    email: 'admin@nextfit.com',
    nome: 'Administrador',
    senha: 'admin123',
    role: 'admin'
});

const options = {
    hostname: 'localhost',
    port: 7777,
    path: '/api/auth/register',
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Content-Length': Buffer.byteLength(data)
    }
};

const req = http.request(options, (res) => {
    let responseData = '';
    res.on('data', (chunk) => {
        responseData += chunk;
    });
    res.on('end', () => {
        if (res.statusCode === 200) {
            console.log('='.repeat(50));
            console.log('✅ Usuário Admin Criado com Sucesso!');
            console.log('='.repeat(50));
            console.log('Email: admin@nextfit.com');
            console.log('Senha: admin123');
            console.log('Role: admin');
            console.log('='.repeat(50));
        } else {
            console.log('Status:', res.statusCode);
            console.log('Response:', responseData);
        }
    });
});

req.on('error', (e) => {
    console.error('Erro:', e.message);
    console.log('Certifique-se de que o backend está rodando na porta 7777');
});

req.write(data);
req.end();
